File: Łatwe/C++/Sort 1/Sort1.cpp
#include <bits/stdc++.h>

int main() {
    int t;
    std::cin >> t;
    for (int i = 0; i < t; i++) {
        int n;
        std::cin >> n;
        std::string names[n];
        int x[n];
        int y[n];
        double distances[n];
        for (int j = 0; j < n; j++) {
            std::cin >> names[j] >> x[j] >> y[j];
            distances[j] = sqrt(x[j] * x[j] + y[j] * y[j]);
        }
        double originalDistances[n];
        std::copy(distances, distances + n, originalDistances);
        std::sort(distances, distances + n);
        for (int j = 0; j < n; j++) {
            for (int k = 0; k < n; k++) {
                if (originalDistances[k] == distances[j])
                    std::cout << names[k] << " " << x[k] << " " << y[k] << std::endl;
            }
        }
    }
}

File: Łatwe/Java/Sort 1/Sort1.java
import java.util.*;

class Sort1 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for (int i = 0; i < t; i++) {
            int n = in.nextInt();
            String[] names = new String[n];
            int[] x = new int[n];
            int[] y = new int[n];
            double[] distances = new double[n];
            for (int j = 0; j < n; j++) {
                names[j] = in.next();
                x[j] = in.nextInt();
                y[j] = in.nextInt();
                distances[j] = Math.sqrt(x[j] * x[j] + y[j] * y[j]);
            }
            double[] originalDistances = distances.clone();
            Arrays.sort(distances);
            for (int j = 0; j < n; j++) {
                for (int k = 0; k < n; k++) {
                    if (originalDistances[k] == distances[j])
                        System.out.println(names[k] + " " + x[k] + " " + y[k]);
                }
            }
        }
    }
}
